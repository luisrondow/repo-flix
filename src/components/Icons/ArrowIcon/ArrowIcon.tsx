const ArrowIcon = ({ direction }: { direction: 'right' | 'left' | 'down' }) => {
  const rotate = direction === 'left' ? 'rotate-180' : direction === 'down' ? 'rotate-90' : ''

  return (
    <svg
      xmlns="http://www.w3.org/2000/svg"
      fill="none"
      viewBox="0 0 24 24"
      strokeWidth={1.5}
      stroke="currentColor"
      className={`${rotate} h-12 w-12 stroke-gray-light`}
    >
      <path strokeLinecap="round" strokeLinejoin="round" d="M8.25 4.5l7.5 7.5-7.5 7.5" />
    </svg>
  )
}

export default ArrowIcon
