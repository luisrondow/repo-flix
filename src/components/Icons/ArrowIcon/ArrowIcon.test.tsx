import { render } from '@testing-library/react'
import ArrowIcon from './ArrowIcon'

describe('ArrowIcon', () => {
  it('should render', () => {
    const { container } = render(<ArrowIcon direction="right" />)

    expect(container).toMatchSnapshot()
  })

  it('should render with direction right', () => {
    const { container } = render(<ArrowIcon direction="left" />)

    expect(container.querySelector('svg')).toHaveClass('rotate-180')
    expect(container).toMatchSnapshot()
  })

  it('should render with direction down', () => {
    const { container } = render(<ArrowIcon direction="down" />)

    expect(container.querySelector('svg')).toHaveClass('rotate-90')
    expect(container).toMatchSnapshot()
  })
})
