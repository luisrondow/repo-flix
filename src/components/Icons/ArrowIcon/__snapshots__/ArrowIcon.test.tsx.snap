// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArrowIcon should render 1`] = `
<div>
  <svg
    class=" h-12 w-12 stroke-gray-light"
    fill="none"
    stroke="currentColor"
    stroke-width="1.5"
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M8.25 4.5l7.5 7.5-7.5 7.5"
      stroke-linecap="round"
      stroke-linejoin="round"
    />
  </svg>
</div>
`;

exports[`ArrowIcon should render with direction down 1`] = `
<div>
  <svg
    class="rotate-90 h-12 w-12 stroke-gray-light"
    fill="none"
    stroke="currentColor"
    stroke-width="1.5"
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M8.25 4.5l7.5 7.5-7.5 7.5"
      stroke-linecap="round"
      stroke-linejoin="round"
    />
  </svg>
</div>
`;

exports[`ArrowIcon should render with direction right 1`] = `
<div>
  <svg
    class="rotate-180 h-12 w-12 stroke-gray-light"
    fill="none"
    stroke="currentColor"
    stroke-width="1.5"
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M8.25 4.5l7.5 7.5-7.5 7.5"
      stroke-linecap="round"
      stroke-linejoin="round"
    />
  </svg>
</div>
`;
